jax/_src/lax/parallel.py

```diff
@@ -479,6 +479,10 @@ def xeinsum(spec: str, *operands):
     raise ValueError("Found subscript(s) "
                      f"{set(out_subs) - set().union(*all_in_subs)} "
                      "appearing in the output spec but not in the input")
+  if len(out_named) != len(set(out_named)):
+    raise ValueError("Output axes should be unique.")
+  if len(out_subs) != len(set(out_subs)):
+    raise ValueError("Output subscripts should be unique.")
```

Comment:
An alternative would be to do something like this if we want a more verbose error message and don't mind iterating over the array
```python
    for out_sub in out_subs:
        if out_subs.count(out_sub) != 1:
            raise ValueError("Output subscript %s appeared more than once in the output." % out_sub))
```